scripts/transitive_reduction_nonimplications.rb

```diff
@@ -0,0 +1,23 @@
+require 'json'
+
+# Given as an input the JSON produced by `lake exe extract_implications --json`, outputs
+# a transitive reduction of the non-implications
+
+require File.join(__dir__, 'graph')
+
+graph = Graph.new
+data = JSON.parse(File.read(ARGV[0]))
+data["implications"].each { |s|
+  graph.add_edge([false, s["rhs"][8,10].to_i], [false, s["lhs"][8,10].to_i])
```

Comment:
Is this `[8,10]` extracting the number from the definition name? But what if the number is longer than two characters? Or is this something else?

Reply:
It takes a substring of length 10 starting from the eighth character

Reply:
Ah, it's length, not end. TIL :-)